reconstruction/cvt/src/main/java/org/jlab/rec/cvt/measurement/Measurements.java

```diff
@@ -425,25 +443,149 @@ else if(type==DetectorType.BMT) {
+        switch (intersectionCount) {
+            case 0:
+                return Double.POSITIVE_INFINITY;
+            case 1:
+                return intersections.get(0).y();
+            case 2:
+                // Choose the intersection closest to the cluster center on the z-axis
+                Point3D closest = getClosestIntersectionToClusterZ(cluster, intersections);
+                return closest.y();
+            default:
+                return Double.POSITIVE_INFINITY;
+        }
+    }
+
+    private double getY(int h, Ray ray, Surface surface) {
+        if (ray == null || surface == null) {
+            return Double.POSITIVE_INFINITY;
+        }
+
+        Line3D line = ray.toLine(); // Shared line for both cylinder and plane
+        if (surface.cylinder != null) {
+            return handleCylinderIntersection(h, surface.cylinder, line);
+        }
+
+        if (surface.plane != null) {
+            return handlePlaneIntersection(line, surface.plane);
+        }
+
+        return Double.POSITIVE_INFINITY; // Default if no surface type matched
     }
     
-    private int getHemisphere(Ray ray, Surface surface){
-        if(surface.cylinder==null)
-            return 0;
-        List<Point3D> trajs = new ArrayList<>();
+    private double handleCylinderIntersection(int h, Cylindrical3D cylinder, Line3D line) {
+        
+        List<Point3D> intersections = new ArrayList<>();
+        int intersectionCount = cylinder.intersection(line, intersections);
+
+        switch (intersectionCount) {
+            case 0:
+                return Double.POSITIVE_INFINITY;
+            case 1:
+                return intersections.get(0).y();
+            case 2:
+                // Choose 
+                double yFirst = intersections.get(0).y();
+                double ySecond =intersections.get(1).y();
+
+            return (int) Math.signum(yFirst)==h ? yFirst : ySecond;
+            
+            default:
+                return Double.POSITIVE_INFINITY;
+        }
+    }
+    
+    private double handlePlaneIntersection(Line3D line, Plane3D plane) {
+        Point3D intersection = new Point3D();
+        if (plane.intersection(line, intersection) != 0) {
+            return intersection.y();
+        }
+        return Double.POSITIVE_INFINITY;
+    }
+
+    private Point3D getClosestIntersectionToClusterZ(Cluster cluster, List<Point3D> intersections) {
+        Point3D first = intersections.get(0);
+        Point3D second = intersections.get(1);
+
+        double distanceToFirst = Math.abs(cluster.center().z() - first.z());
+        double distanceToSecond = Math.abs(cluster.center().z() - second.z());
+
+        return distanceToFirst < distanceToSecond ? first : second;
+    }
+
+    private int getHemisphere(Cluster cluster, Ray ray, Surface surface){
+        double Y = this.getY(cluster, ray, surface);
+        if(Y == Double.POSITIVE_INFINITY) return 0;
+        surface.rayYinterc=Y;
+        return (int) Math.signum(Y);
```

Comment:
if misalignments are large in y, the sign of the y coordinate of the intersection point doesn't really correspond to the hemisphere